pkg/collector/prometheus.go

```diff
@@ -0,0 +1,594 @@
+		defer wg.Done()
+		for pkgID, val := range nodeEnergy.EnergyInDRAM.Stat {
+			ch <- prometheus.MustNewConstMetric(
+				c.nodeDesc.nodeDramJoulesTotal,
+				prometheus.CounterValue,
+				float64(val.Aggr)/miliJouleToJoule,
+				pkgID, nodeName, "rapl",
+			)
+		}
+		for pkgID, val := range nodeEnergy.EnergyInPkg.Stat {
+			ch <- prometheus.MustNewConstMetric(
+				c.nodeDesc.nodePackageJoulesTotal,
+				prometheus.CounterValue,
+				float64(val.Aggr)/miliJouleToJoule,
+				pkgID, nodeName, "rapl",
+			)
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ch <- prometheus.MustNewConstMetric(
+			c.nodeDesc.nodePlatformJoulesTotal,
+			prometheus.CounterValue,
+			float64(nodeEnergy.SensorEnergy.Aggr())/miliJouleToJoule,
+			nodeName, "acpi",
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.nodeDesc.nodeOtherComponentsJoulesTotal,
+			prometheus.CounterValue,
+			float64(nodeEnergy.EnergyInOther.Aggr())/miliJouleToJoule,
+			nodeName,
+		)
+		if config.EnabledGPU {
+			for gpuID, val := range nodeEnergy.EnergyInGPU.Stat {
+				ch <- prometheus.MustNewConstMetric(
+					c.nodeDesc.nodeGPUJoulesTotal,
+					prometheus.CounterValue,
+					float64(val.Aggr)/miliJouleToJoule,
+					gpuID, nodeName, "nvidia",
+				)
+			}
+		}
+	}()
+}
+
+// updatePodMetrics send pod metrics to prometheus
+func (c *PrometheusCollector) UpdatePodMetrics(wg *sync.WaitGroup, ch chan<- prometheus.Metric) {
+	for _, container := range containersMetrics {
+		wg.Add(1)
+		go func(container *ContainerMetrics) {
+			defer wg.Done()
+			// TODO: After removing this metric in the next release, we need to refactor and remove the ToPrometheusValues function
+			podEnergyStatusLabelValues := []string{container.PodName, container.ContainerName, container.Namespace, container.Command}
+			for _, label := range podEnergyStatLabels[4:] {
+				val := container.ToPrometheusValue(label)
+				podEnergyStatusLabelValues = append(podEnergyStatusLabelValues, val)
+			}
+			ch <- prometheus.MustNewConstMetric(
+				c.podDesc.podEnergyStat,
+				prometheus.GaugeValue,
+				float64(container.Curr()),
+				podEnergyStatusLabelValues...,
+			) // deprecated metric
+		}(container)
+
+		wg.Add(1)
+		go func(container *ContainerMetrics) {
+			defer wg.Done()
+			for cpu, cpuTime := range container.CurrCPUTimePerCPU {
+				ch <- prometheus.MustNewConstMetric(
+					c.containerDesc.containerCPUTime,
+					prometheus.GaugeValue,
+					float64(cpuTime),
+					container.PodName, container.ContainerName, container.Namespace, strconv.Itoa(int(cpu)),
+				) // deprecated metric
+			}
+		}(container)
+
+		wg.Add(1)
+		go func(container *ContainerMetrics) {
+			defer wg.Done()
+			ch <- prometheus.MustNewConstMetric(
+				c.containerDesc.containerCoreJoulesTotal,
+				prometheus.CounterValue,
+				float64(container.EnergyInCore.Aggr)/miliJouleToJoule,
+				container.PodName, container.ContainerName, container.Namespace, container.Command[:int(math.Min(float64(len(container.Command)), 10))],
```

Comment:
`container.Command[:int(math.Min(float64(len(container.Command)), 10))` should we only calculate it once?

Reply:
and 10 should be a const?

Reply:
```
const commandLenLimit=10
var _command := container.Command
if len(_command) > commandLenLimit {
_command = container.Command[:commandLenLimit]
}
```

Reply:
Yes, sounds good